frontends/infinite-corridor/src/api/learning-resources/hooks.ts

```diff
@@ -1,9 +1,34 @@
-import type { LearningResourceResult } from "ol-search-ui"
-import { useQuery } from "react-query"
-import * as urls from "./urls"
+import type {
+  LearningResource,
+  PaginatedUserListItems,
+  UserList
+} from "ol-search-ui"
+import type { PaginatedResult, PaginationSearchParams } from "ol-util"
+import axios from "../../libs/axios"
+import { useQuery, UseQueryResult } from "react-query"
+import { urls, keys, UserListOptions } from "./urls"
 
 const useResource = (type: string, id: number) => {
-  return useQuery<LearningResourceResult>(urls.resource(type, id))
+  const url = urls.resourceDetails(type, id)
+  const key = keys.resourceDetails(type, id)
```

Comment:
React query requires a query key for all queries: `useQuery(key, fetcher?)`. Previously we were using API urls as string-based keys and relying on a default fetcher specified in our overall react query configuration.

The alternative (which I've switched to here) is array (`string[]`) keys. _We could still use a default fetcher, but I decided its not worth the effort._

Array-based keys have two advantages:
- they make invalidating caches easier. E.g., if you have several queries with keys `["userlist", "details", 5]`, `["userlist", "listing"]`, `["userlist", "something else"]`, then you can invalidate all those keys by calling `queryClient.invalidateQueries(["userlist"])`. Ie, react-query treats array-based query keys as something like a directory structure.
- we use react query 3, but react-query 4 only supports array-based keys